pkg/intermediate/aggregate.go

```diff
@@ -984,3 +993,29 @@ func isCorrelationRequired(flowType uint8, record entities.Record) bool {
 	}
 	return false
 }
+
+func fillHttpVals(incomingHttpVals, existingHttpVals string) (string, error) {
+	incomingHttpValsJson := make(map[int32]string)
+	existingHttpValsJson := make(map[int32]string)
+
+	if incomingHttpVals != "" {
+		if err := json.Unmarshal([]byte(incomingHttpVals), &incomingHttpValsJson); err != nil {
+			return "", fmt.Errorf("Error parsing JSON: %v", err)
+		}
+	}
+	if existingHttpVals != "" {
+		if err := json.Unmarshal([]byte(existingHttpVals), &existingHttpValsJson); err != nil {
+			return "", fmt.Errorf("Error parsing JSON: %v", err)
+		}
+	}
+	if len(existingHttpValsJson) > 0 {
+		for key, value := range existingHttpValsJson {
+			incomingHttpValsJson[key] = value
```

Comment:
Sorry for my ignorance, I'm a little bit confused about what we are trying to do here, why we fill value of `existingHttpValsJson` to `incomingHttpValsJson` if we have already unmarshal the `incomingHttpVals` in L1001

Reply:
This step is for deduplicating any keys present in both jsons and adding new keys not present in the incomingHttpValsJson